rayzardemo.py

```diff
@@ -97,12 +104,35 @@ def textzng(zng):
 
     ray.init()
 
-    zar = Archive.remote(args.zarroot)
-    srcs = ray.get(zar.find.remote(args.find))
-    xformed = [zq.remote(args.filter, src) for src in srcs]
+    # Launch a ray actor to search for the log chunks that match the query.
+    search = ArchiveSearch.remote(args.zarroot, args.find)
 
+    # Launch a ray actor to aggregate all of the matching chunks.
     agg = Aggregator.remote()
-    res = ray.get(agg.aggregate.remote(args.merge, xformed))
+
+    maxtasks = 4 # XXX
+    tasks = []
+
+    # Loop over the search results and do an parallel aggregation up to
+    # maxtasks in parallel.
+    while True:
+        location = ray.get(search.next.remote())
+        if location == None:
+            break
+        # For each hit, we launch an actor to read the chunk and filter
+        # to only the events that match the search.
```

Comment:
```suggestion
        # For each hit, we create a task to read the chunk and filter
        # to only the events that match the search.
```

Reply:
But it's creating a Ray actor, which is stateful, not a stateless Ray task... ?

Reply:
My understanding of ray is obviously limited so far, but my interpretation of their docs is that the line above where you perform `agg = Aggregator.remote()` is that it represents 'launching an actor'; however, here you are calling a method on that existing actor, not launching a new one?

Reply:
I stand corrected.  It is a stateless loading of the file the corresponds each generation of a path from the actor that is searching.  I'll fix the comment, thanks!